frame/staking/src/pallet/impls.rs

```diff
@@ -405,7 +405,7 @@ impl<T: Config> Pallet<T> {
 		start_session_index: SessionIndex,
 		is_genesis: bool,
 	) -> Option<Vec<T::AccountId>> {
-		let (election_result, weight) = if is_genesis {
+		let election_result = if is_genesis {
 			T::GenesisElectionProvider::elect().map_err(|e| {
```

Comment:
note: `elect` is a self-weighing function now as well.

Reply:
Should we add something to the `ElectionProvider` trait saying that we expect `elect` to be self-weighing? Optimally we could enforce that programatically or with a test bc just a doc comment could easily become stale & then misleading

Reply:
Same goes for `voters` and `targets`

Reply:
at least as a doc, it is already mentioned for all of them: https://github.com/paritytech/substrate/pull/9569/files#diff-cf4011c50fccf927439dea616af75d3bf593ad8e244c2c9b06a3364162f50244R283